WooCommerce/Classes/ViewModels/InAppPurchases/UpgradesViewModel.swift

```diff
@@ -229,6 +202,23 @@ final class UpgradesViewModel: ObservableObject {
         }
     }
 
+    /// Checks whether the current user is the site owner, as only the site owner can perform
+    /// In-App Purchases upgrades, despite their site role
+    ///
+    func checkForSiteOwnership() {
+        if let site = stores.sessionManager.defaultSite, !site.isSiteOwner {
+            self.upgradeViewState = .prePurchaseError(.userNotAllowedToUpgrade)
+        } else {
+            Task {
+                await fetchViewData()
+            }
+        }
```

Comment:
We can tweak this next week, but it doesn't follow to me that `checkForSiteOwnership` also fetches the view data.

Reply:
Definitely, logged here: https://github.com/woocommerce/woocommerce-ios/issues/10152